Description: raise SkipTest if LaTeX (or a required LaTeX package) is not found
 If LaTeX (or a required LaTeX package) is not found when running
 test_build_latex, raise SkipTest instead of just printing a message to stderr.
Author: Jakub Wilk <jwilk@debian.org>
Bug: https://bitbucket.org/birkenfeld/sphinx/issue/896/test_build_latex-could-use-skiptest
Forwarded: no
Last-Update: 2012-03-13

--- a/tests/test_build_latex.py
+++ b/tests/test_build_latex.py
@@ -18,6 +18,7 @@
 from sphinx.writers.latex import LaTeXTranslator
 
 from util import *
+from util import SkipTest
 from test_build_html import ENV_WARNINGS
 
 
@@ -68,17 +69,17 @@
             return True
 
     if kpsetest('article.sty') is None:
-        print >>sys.stderr, \
-              'info: not running latex, it doesn\'t seem to be installed'
-        return
+        raise SkipTest(
+              'not running latex, it doesn\'t seem to be installed'
+        )
     for filename in ['fancyhdr.sty', 'fancybox.sty', 'titlesec.sty',
                      'amsmath.sty', 'framed.sty', 'color.sty', 'fancyvrb.sty',
                      'threeparttable.sty']:
         if not kpsetest(filename):
-            print >>sys.stderr, \
-                  'info: not running latex, the %s package doesn\'t ' \
+            raise SkipTest(
+                  'not running latex, the %s package doesn\'t ' \
                   'seem to be installed' % filename
-            return
+            )
 
     # now, try to run latex over it
     cwd = os.getcwd()
